mloop/visualizations.py

```diff
@@ -197,30 +198,64 @@ def plot_cost_vs_run(self):
         for ut in self.unique_types:
             artists.append(plt.Line2D((0,1),(0,0), color=_color_from_controller_name(ut), marker='o', linestyle=''))
         plt.legend(artists,self.unique_types,loc=legend_loc)
+    
+    def _ensure_parameter_subset_valid(self, parameter_subset):
+        # Make sure that parameter_subset is iterable.
+        if not hasattr(parameter_subset, '__iter__'):
+            message = 'parameter_subset should be a list, or list-like.'
+            self.log.error(message)
+            raise ValueError(message)
+            
+        # Ensure all indices are valid.
+        for ind in parameter_subset:
+            if ind not in self.param_numbers:
+                message = '{ind} is not a valid parameter index.'.format(ind=ind)
+                self.log.error(message)
+                raise ValueError(message)
+        
         
-    def plot_parameters_vs_run(self):
+    def plot_parameters_vs_run(self, parameter_subset=None):
         '''
         Create a plot of the parameters versus run number.
+    
+        Args:
+            parameter_subset (list-like): The indices of parameters to plot. The
+                indices should be 1-based, i.e. the first parameter is
```

Comment:
I'm a little suspicious of any 1-based list, but as the one who's used this package more recently I'll trust you that this fits more naturally

Reply:
Ah funny timing. I was in the middle of writing my response below when you posted this, and I didn't see yours until I had already submitted mine. Yeah I'm not a huge fan of 1-indexing in a 0-indexed language but I do really think that the legend and user-provided arguments should be indexed in the same way.

Reply:
My vote would just be to 0-index everywhere, if that won't break anything too badly

Reply:
Another option would be to change the legend to index parameters starting at 0. In that case it would probably be best to change that for all of the visualizers though so that the indices agree when comparing plots of the same optimization using different learners/visualizers.

Reply:
Oh just noticed your comment before mine. I need to get in the habit or refreshing the page before commenting. Yeah I think 0-indexing the is simpler and more elegant solution so I'll go for that. Thanks for your input.

Reply:
Resolved by commits 68ea4cc, 247518f, fe3ed0e, and afb1f34.